tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -15320,17 +15421,23 @@ void TryAddShaderAttrFromTargetProfile(Sema &S, FunctionDecl *FD) {
       HLSLShaderAttr::CreateImplicit(S.Context, fullName);
 
   FD->addAttr(pShaderAttr);
+  isActiveEntry = true;
   return;
 }
 
 void DiagnoseEntry(Sema &S, FunctionDecl *FD) {
-  TryAddShaderAttrFromTargetProfile(S, FD);
+  bool isActiveEntry = false;
```

Comment:
A comment explaining `isActiveEntry` is needed, along with a TODO for keying it off of the exports for lib target.  I see that `TryAddShaderAttrFromTargetProfile` sets this, but we should check if it's a lib here and skip calling that entirely if so.  For example:

```cpp
if (S.getLangOpts().IsHLSLLibrary) {
  // TODO: Analyze -exports option to determine which entries
  // are active for lib target.
  // For now, assume all entries are active.
  isActiveEntry = true;
} else {
  TryAddShaderAttrFromTargetProfile(S, FD, isActiveEntry);
}
```

Note that we are setting `isActiveEntry` before we have actually confirmed that this is an entry via the shader attribute.  That's ok, as long as we keep that in mind.